### PR TITLE
deprecate usm configuration values

### PR DESCRIPTION
**Pull request by guyarb**
### What does this PR do?
The purpose of this pull request is to modify the configuration value `network_config.{http_replace_rules|max_http_stats_buffered|max_tracked_http_connections}` and replace it with `service_monitoring_config.{http_replace_rules|max_http_stats_buffered|max_tracked_http_connections}`, while ensuring backward compatibility.

The change is based on the changes we've made in #16854

### Motivation
The reason for this pull request is to support a larger refactoring effort aimed at migrating all relevant configuration values from the `network_config` tree to the `service_monitoring_config` tree.

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/config/system_probe.go [hunk 1]
```diff
@@ -188,19 +188,30 @@ func InitSystemProbeConfig(cfg Config) {
...
-	cfg.SetEnvKeyTransformer(httpRules, func(in string) interface{} {
-		var out []map[string]string
-		if err := json.Unmarshal([]byte(in), &out); err != nil {
-			log.Warnf(`%q can not be parsed: %v`, httpRules, err)
+
+	oldHTTPRules := join(netNS, "http_replace_rules")
+	newHTTPRules := join(smNS, "http_replace_rules")
+	cfg.BindEnv(newHTTPRules)
+	cfg.BindEnv(oldHTTPRules, "DD_SYSTEM_PROBE_NETWORK_HTTP_REPLACE_RULES")
+	httpRulesTransformer := func(key string) func(in string) interface{} {
```
Comment (val06) [inline pkg/config/system_probe.go:198]: introduce an alias for better readability:

`type transformerFunction func(in string) interface{}`

**Review by val06 — CHANGES_REQUESTED** [review]
cred

File: pkg/config/system_probe.go (hunk 1 above)
Comment (guyarb) [reply pkg/config/system_probe.go:198]: done

**Review by val06 — APPROVED** [review]

**Review by hmahmood — APPROVED** [review]

File: releasenotes/notes/deprecating-usm-configuration-values-6c43a0181c2cc821.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+---
+features:
+  - The `network_config.http_replace_rules` configuration has changed to `service_monitoring_config.http_replace_rules`.
+  - The `network_config.max_tracked_http_connections` configuration has changed to `service_monitoring_config.max_tracked_http_connections`.
+  - The `network_config.max_http_stats_buffered` configuration has changed to `service_monitoring_config.max_http_stats_buffered`.
+deprecations:
+  - The `network_config.http_replace_rules` configuration is now deprecated. Use `service_monitoring_config.http_replace_rules` instead.
+  - The `network_config.max_tracked_http_connections` configuration is now deprecated. Use `service_monitoring_config.max_tracked_http_connections` instead.
+  - The `network_config.max_http_stats_buffered` configuration is now deprecated. Use `service_monitoring_config.max_http_stats_buffered` instead.
```
Comment (urseberry) [inline releasenotes/notes/deprecating-usm-configuration-values-6c43a0181c2cc821.yaml:9]:
```suggestion
  - The `network_config.http_replace_rules` configuration has been deprecated. Use `service_monitoring_config.http_replace_rules` instead.
  - The `network_config.max_tracked_http_connections` configuration has been deprecated. Use `service_monitoring_config.max_tracked_http_connections` instead.
  - The `network_config.max_http_stats_buffered` configuration has been deprecated. Use `service_monitoring_config.max_http_stats_buffered` instead.
```
Recommend keeping verb tense consistent throughout the release note.

**Review by urseberry — APPROVED** [review]
Approving because my feedback is minor and doesn't need to hold up the process.

File: releasenotes/notes/deprecating-usm-configuration-values-6c43a0181c2cc821.yaml (hunk 2 above)
Comment (guyarb) [reply releasenotes/notes/deprecating-usm-configuration-values-6c43a0181c2cc821.yaml:9]: thanks, committed
I copied it from a different release note https://github.com/DataDog/datadog-agent/blob/main/releasenotes/notes/enable-http-monitoring-config-value-21ae9f834e3e2c67.yaml

File: pkg/network/config/config_test.go [hunk 3]
```diff
@@ -444,9 +444,51 @@ func TestHTTPReplaceRules(t *testing.T) {
...
+			assert.Equal(t, r, cfg.HTTPReplaceRules[i])
+		}
+	})
+
+	t.Run("via deprecated ENV variable", func(t *testing.T) {
+		newConfig(t)
+		t.Setenv("DD_SYSTEM_PROBE_NETWORK_HTTP_REPLACE_RULES", envContent)
+
+		_, err := sysconfig.New("")
+		require.NoError(t, err)
```
Comment (brycekahle) [inline pkg/network/config/config_test.go:480]: If you are just testing with environment variables, you shouldn't need to call `sysconfig.New` now.

**Pull request merged**